hack/golangci.yaml

```diff
@@ -173,127 +127,181 @@ linters:
+            contextual k8s.io/client-go/tools/record/.*
+            contextual k8s.io/component-helpers/.*
+            contextual k8s.io/cri-api/.*
+            contextual k8s.io/cri-client/.*
+            contextual k8s.io/csi-translation-lib/.*
+            contextual k8s.io/dynamic-resource-allocation/.*
+            contextual k8s.io/endpointslice/.*
+            contextual k8s.io/kms/.*
+            contextual k8s.io/kube-controller-manager/.*
+            contextual k8s.io/kube-proxy/.*
+            contextual k8s.io/kube-scheduler/.*
+            contextual k8s.io/sample-apiserver/.*
+            contextual k8s.io/sample-cli-plugin/.*
+            contextual k8s.io/sample-controller/.*
+            contextual k8s.io/kubernetes/cmd/kube-proxy/.*
+            contextual k8s.io/kubernetes/cmd/kube-scheduler/.*
+            contextual k8s.io/kubernetes/pkg/controller/.*
+            contextual k8s.io/kubernetes/pkg/scheduler/.*
+            contextual k8s.io/kubernetes/test/e2e/dra/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/cm/dra/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/pleg/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/clustertrustbundle/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/token/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/cadvisor/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/oom/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/status/.*
+            contextual k8s.io/kubernetes/pkg/kubelet/sysctl/.*
+            
+            # As long as contextual logging is alpha or beta, all WithName, WithValues,
+            # NewContext calls have to go through klog. Once it is GA, we can lift
+            # this restriction. Whether we then do a global search/replace remains
+            # to be decided.
+            with-helpers .*
+    depguard:
+      rules:
+        main:
+          files:
+            - $all
+            - "!$test"
+            - "!**/test/**"
+          deny:
+            - pkg: "github.com/google/go-cmp/cmp"
+              desc: "cmp is allowed only in test files"
+    forbidigo:
+      analyze-types: true
+      forbid:
+        - pattern: ^managedfields\.ExtractInto$
+          pkg: ^k8s\.io/apimachinery/pkg/util/managedfields$
+          msg: should not be used because managedFields was removed
+        - pattern: \.Extract
+          pkg: ^k8s\.io/client-go/applyconfigurations/
+          msg: should not be used because managedFields was removed
+        - pattern: \.Add$
+          pkg: ^k8s\.io/component-base/featuregate$         
+          msg: should not use Add, use AddVersioned instead
+    gocritic:	
+      enabled-checks:	
+        - equalFold	
+        - boolExprSimplify
+    revive:
+      # Only these rules are enabled.
+      rules:
+        - name: exported
+          arguments:
+            - disableStutteringCheck
+    staticcheck:
+      checks:
+        - "all"
+        - "-QF1001"  # Apply De Morgan’s law
```

Comment:
I guess we can incrementally identify some of these to eliminate, many sound very opinionated, some of them sound like possible bugs.